tests/javaRosa.js

```diff
@@ -522,6 +526,40 @@ require([
                                    "wrong <help> node count\n" + xml);
             });
         });
+
+        it("should rename itext item ID after move", function () {
+            util.loadXML("");
+            util.addQuestion("Select", "ns");
+            util.addQuestion("Select", "ew");
+            var north = util.getMug("ns/item1"),
+                south = util.getMug("ew/item1");
+            north.p.defaultValue = "north";
+            south.p.defaultValue = "south";
+            north.form.moveMug(south, north, "after");
+            util.assertXmlEqual(util.call("createXML"), ITEXT_ITEM_RENAME_XML,
+                                {normalize_xmlns: true});
+        });
+
+        it("should rename group's child itext item IDs after move group", function () {
+            util.loadXML("");
+            var green = util.addQuestion("Group", "green"),
+                blue = util.addQuestion("Group", "blue");
+            util.addQuestion("Text", "text");
+            blue.form.moveMug(blue, green, "before");
+            util.assertXmlEqual(util.call("createXML"),
+                                ITEXT_ITEM_RENAME_GROUP_MOVE_XML,
+                                {normalize_xmlns: true});
+        });
+
+        it("should not auto-update itext ID when multiple questions point to auto-ish-id", function () {
+            util.loadXML(ITEXT_ITEM_NON_AUTO_ID_XML);
+            var north = util.getMug("north");
+            north.p.nodeID = "west";
+            var xml = call("createXML"),
+                $xml = $(xml);
+            assert.strictEqual($xml.find("text#north-label").length, 2,
+                               "wrong <text> node count\n" + xml);
+        });
```

Comment:
A few new tests to verify that the itext-auto-id bugs are fixed. Some of this code came from https://github.com/dimagi/Vellum/compare/154742-auto-itext-ids, a branch that I had started this javaRosa refactor a while ago.